bson/src/test/unit/org/bson/codecs/BigDecimalCodecSpecification.groovy

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.codecs
+
+import org.bson.BsonDocument
+import org.bson.BsonDocumentReader
+import org.bson.BsonDocumentWriter
+import org.bson.BsonReader
+import spock.lang.Specification
+
+class BigDecimalCodecSpecification extends Specification {
+
+    def 'should round trip BigDecimal successfully'(BigDecimal expected) {
+        given:
+        BigDecimalCodec codec = new BigDecimalCodec()
+
+        def writer = new BsonDocumentWriter(new BsonDocument())
+        writer.writeStartDocument()
+        writer.writeName('bigDecimal')
+        codec.encode(writer, expected, EncoderContext.builder().build())
+        writer.writeEndDocument()
+
+        BsonReader bsonReader = new BsonDocumentReader(writer.getDocument())
+        bsonReader.readStartDocument()
+        bsonReader.readName()
+
+        when:
+        BigDecimal actual = codec.decode(bsonReader, DecoderContext.builder().build())
+
+        then:
+        expected == actual
```

Comment:
There should be two assertions: one that the BigDecimal encodes _to_ a Decimal128 and another that it decodes _from_ a Decimal128.  Otherwise it would be possible for the codec to be encoding to/decoding from a BSON double in some cases and the assertion would still pass.